berkdb/rep/rep_record.c

```diff
@@ -469,12 +510,22 @@ __rep_process_message(dbenv, control, rec, eidp, ret_lsnp, commit_gen)
 	 * to get in sync.
 	 */
 	if (rp->gen < gen && rp->rectype != REP_ALIVE_REQ &&
-	    rp->rectype != REP_NEWCLIENT && rp->rectype != REP_MASTER_REQ &&
-	    rp->rectype != REP_DUPMASTER) {
+		rp->rectype != REP_NEWCLIENT && rp->rectype != REP_MASTER_REQ) {
```

Comment:
This is a change to the election state machine: a master from a previous generation shouldn't be allowed to downgrade the current master.